pandas/core/frame.py

```diff
@@ -2954,7 +2954,7 @@ def __getitem__(self, key):
             return self.where(key)
 
         # Do we have a (boolean) 1d indexer?
-        if com.is_bool_indexer(key):
+        if com.is_bool_indexer(key) or isinstance(key, Series) and key.empty:
```

Comment:
can you put parens around the and clause to make this obvious

Reply:
Done. Had to adjust the condition a bit to cover np.array